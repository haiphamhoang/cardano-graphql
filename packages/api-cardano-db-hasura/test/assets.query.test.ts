/* eslint-disable camelcase */
import path from 'path'

import { DocumentNode } from 'graphql'
import util from '@cardano-graphql/util'
import { TestClient } from '@cardano-graphql/util-dev'
import { buildClient } from './util'
import { Genesis } from '@src/graphql_types'

const genesis = {
  byron: require('../../../config/network/mainnet/genesis/byron.json'),
  shelley: require('../../../config/network/mainnet/genesis/shelley.json')
} as Genesis

function loadQueryNode (name: string): Promise<DocumentNode> {
  return util.loadQueryNode(path.resolve(__dirname, '..', 'src', 'example_queries', 'assets'), name)
}

describe('assets', () => {
  let client: TestClient
  beforeAll(async () => {
    client = await buildClient('http://localhost:3100', 'http://localhost:8090', 5442, genesis)
  })

  it('can return information on assets', async () => {
    const result = await client.query({
      query: await loadQueryNode('assets')
    })
    const { assets_aggregate, assets } = result.data
    const { aggregate } = assets_aggregate
    expect(aggregate.count).toBeDefined()
    expect(assets).toMatchSnapshot()
  })
})
