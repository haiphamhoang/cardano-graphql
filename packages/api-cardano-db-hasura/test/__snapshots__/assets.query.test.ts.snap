// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assets can return information on assets 1`] = `Array []`;
